src/core/mcis/monitor.go

```diff
@@ -149,31 +178,24 @@ func InstallMonitorAgentToMcis(nsId string, mcisId string, req *McisCmdReq) (Age
 	method := "POST"
 
 	for _, v := range vmList {
-		wg.Add(1)
-
-		vmId := v
-		vmIp := GetVmIp(nsId, mcisId, vmId)
-
-		//cmd := req.Command
-
-		// userName, sshKey := GetVmSshKey(nsId, mcisId, vmId)
-		// if (userName == "") {
-		// 	userName = req.User_name
-		// }
-		// if (userName == "") {
-		// 	userName = sshDefaultUserName
-		// }
-
-		// find vaild username
-		userName, sshKey := GetVmSshKey(nsId, mcisId, vmId)
-		userNames := []string{SshDefaultUserName01, SshDefaultUserName02, SshDefaultUserName03, SshDefaultUserName04, userName, req.User_name}
-		userName = VerifySshUserName(vmIp, userNames, sshKey)
-
-		fmt.Println("[SSH] " + mcisId + "/" + vmId + "(" + vmIp + ")" + "with userName:" + userName)
-		fmt.Println("[CMD] " + cmd)
-
-		go CallMonitoringAsync(&wg, nsId, mcisId, vmId, vmIp, userName, sshKey, method, cmd, &resultArray)
-
+		vmObjTmp, _ := GetVmObject(nsId, mcisId, v)
+		fmt.Println("MonAgentStatus : " + vmObjTmp.MonAgentStatus)
+		
+		if(vmObjTmp.MonAgentStatus != "installed"){
+
+			vmId := v
+			vmIp := GetVmIp(nsId, mcisId, vmId)
+	
+			// find vaild username
+			userName, sshKey := GetVmSshKey(nsId, mcisId, vmId)
+			userNames := []string{SshDefaultUserName01, SshDefaultUserName02, SshDefaultUserName03, SshDefaultUserName04, userName, req.User_name}
```

Comment:
우선순위를 어떻게 하는 것이
실제 소요 시간을 더 적게 할 수 있는지
직접 실험해 보지는 않아서 모르겠지만..

아래와 같은 안도 있겠습니다.

```suggestion
			userNames := []string{userName, req.User_name, SshDefaultUserName01, SshDefaultUserName02, SshDefaultUserName03, SshDefaultUserName04}
```

Reply:
@jihoon-seo 반영해보도록 하겠습니다. 아마도 제시하신 방식이 조금이라도 더 좋은 성능을 보일 것 같네요.

Reply:
반영하였습니다.